internal/updater/utils/html/parsing.go

```diff
@@ -0,0 +1,70 @@
+package utils
+
+import (
+	"errors"
+
+	"github.com/andybalholm/cascadia"
+	"golang.org/x/net/html"
+)
+
+var (
+	ErrAttrNotFound = errors.New("matching attribute not found")
+)
+
+func GetText(n *html.Node) string {
+	return n.FirstChild.Data
+}
+
+func GetAttr(n *html.Node, key string) (string, error) {
+	for _, attr := range n.Attr {
+		if attr.Key == key {
+			return attr.Val, nil
+		}
+	}
+
+	return "", ErrAttrNotFound
+}
+
+func CheckID(n *html.Node, id string) bool {
+	if n.Type == html.ElementNode {
+		s, err := GetAttr(n, "id")
+
+		if err == nil && s == id {
+			return true
+		}
+	}
+
+	return false
+}
```

Comment:
- Add named return to know a bit more what that function does (we want to check the html node has the id matching our id)
- Return early for less indentations
- rename `s` to `idValue`
- rename `n` to `node`
- Return `match` as boolean expression `err == nil && idValue == id` directly

```suggestion
func CheckID(node *html.Node, id string) (match bool) {
	if node.Type != html.ElementNode {
		return false
	}
	idValue, err := GetAttr(node, "id")
	return err == nil && idValue == id
}
```